### DISCUSSION TITLE
Script runs without error but new outlook does not receive any signatures.

### DISCUSSION THREAD

**Discussion by arjentromp** [Set-OutlookSignatures/Set-OutlookSignatures#129] [Q&A] [answered]
Hello!

I am currently in the process of implementing Set-OutlookSignatures within our own hybrid environment. Our mailboxes are hosted exclusively in the cloud. I have a test environment (hybrid aswell) in which i followed the quick setup guide. For these tests i am running without a benefactor license that we may or may not end up purchasing later. Here are the steps that i followed.

1. Registered an Entra ID app using the 'https://login.microsoftonline.com/organizations/adminconsent?client_id=beea8249-8c98-4c76-92f6-ce3c468a61e6' link. (We will probably register an app manually when we move to production)
2. Downloaded Set-OutlookSignatures on a test client that is running windows 10 and is connected to a test domain.
3. Unblocked the script and ran it using the following command: "powershell.exe -noexit -file "C:\Users\admin-arjen\Documents\Set-OutlookSignatures_v4.15.0\Set-OutlookSignatures.ps1" -GraphOnly true -Verbose"

Output can be found below:
[Set-OutlookSignatures_log.txt](https://github.com/user-attachments/files/17589465/Set-OutlookSignatures_log.txt)

Script ran without any errors as far as i can tell. When i went to check the result in classic outlook, the test signatures had been implemented correctly but when i switched to new outlook no signatures had been added. So the local signatures and folders are created correctly in the appdata folder but not in the exchange online mailboxes.

I checked the documentation and i have a feeling it has to do with roaming signatures. But i am not sure as i do not fully understand the difference, if any, between normal signatures hosted in exchange online mailboxes and roaming signatures. I also do not know if new outlook is supported in the free open source version of Set-OutlookSignatures.

The last time i implemented uniform signatures, new outlook was not yet a thing and i could just create a VB script that read userdata straight from AD and generated an HTML in the signatures folder. So my experience is rather outdated and limited in this regard.

I would love some pointers so i can move onto implementing this using intune.

Thanks in advance!

Arjen

**Comment by GruberMarkus** [comment, marked as answer]
Hi @arjentromp,

your assumption is correct: New Outlook uses roaming signatures, not local signatures.

Support for roaming signatures (and syncing them with local signatures) is included in the Benefactor Circle add-on. Read about the features available exclusively to Benefactor Circle members [in the Set-OutlookSignatures repository](https://github.com/Set-OutlookSignatures/Set-OutlookSignatures/blob/main/docs/Benefactor%20Circle.md) and at our partner [ExplicIT Consulting](https://explicitconsulting.at/open-source/set-outlooksignatures/).

Please let me know if this answers your question and if this comment can be marked as answer.

**Reply by arjentromp** [reply]
Thanks for the quick reply!

And yes, it does answer my question. I was under the impression that there were at least 2 kinds of cloud signatures; ones that are stored in exchange online and "roaming signatures". But apparently they are one and the same. That is good to know!

However, does that not mean that new outlook is only supported if you own a Benefactor Circle license?

I was led to believe that new outlook was also supported in the free open source version after reading the following information under the "Overview and features" chapter:

_Applied to all mailboxes (including shared mailboxes¹), specific mailbox groups, specific email addresses (including alias and secondary addresses), or specific user or mailbox properties, for every mailbox across all Outlook profiles (Outlook, New Outlook, Outlook Web), including automapped and additional mailboxes¹_

Please note that New Outlook (and Outlook Web for that matter) does not have a footnote, whereas shared mailboxes does.

Am i reading this wrong or is this an oversight? I don't mean to nitpick, i just want to point out that it could be misinterpreted.

Anyway, thanks for the rapid response! Thank you for your effort and generosity in developing and maintaining free and open source software!

Arjen

**Comment by GruberMarkus** [comment]
The README has the following hint:
```
Set in Outlook Web¹ for the currently logged-in user, including mirroring signatures to the cloud as roaming signatures¹ (Linux/macOS/Windows, Classic and New Outlook)
```

You are correct. The possible misunderstanding will be corrected in the next release. Thanks for pointing this out!

**Reply by arjentromp** [reply]
No problem, glad that I could be of some use.

Thank you for your help.

**Comment by GruberMarkus** [comment]
My pleasure!
